bin/main.html

```diff
@@ -1615,54 +1615,98 @@ <h2>
 };
 on('clicked:undotoproll', undoTopRoll);
 
+async function consumePower(powerConsumption) {
+  const powerConsumptionValue = Number(powerConsumption) ?? 0;
+  if (powerConsumptionValue <= 0) {
+    return;
+  }
+  return new Promise((resolve) => {
+    getAttrs(['EP_t'], function (values) {
+      const currentPower = Number(values['EP_t']);
+      const newPower = currentPower - powerConsumptionValue;
+      if (newPower < 0) {
+        console.log(`You are weak and lack the PWR to do that. Current power: ${currentPower}, consumption: ${powerConsumptionValue}`);
+        throw new Error('You are weak and lack the PWR to do that.');
+      }
+      console.log(`Consuming ${powerConsumptionValue} power. Current power: ${currentPower}, new power: ${newPower}`);
+      setAttrs({ EP_t: newPower }, null, resolve);
+    });
+  });
+}
+
 const EP_ROLL_OPTIONS = [{
   actionId: 'epRoll',
   attributes: [],
-  getChatMessage: () => {
+  getChatMessage: async () => {
```

Comment:
We should move to async functions and promises.
It will clean up callback hell.
These make it a lot easier to make code linear and is supported by all browsers.

Reply:
For clarity - the support has been around for 10 years as part of ES2015